impacted-test-engine/src/main/java/com/teamscale/test_impacted/engine/ImpactedTestEngine.java

```
@@ -44,86 +24,23 @@ public String getId() {
-	}
-
-	private void runTestExecutor(ExecutionRequest request) {
-		List<TestDetails> availableTests = new ArrayList<>();
-		List<TestExecution> testExecutions = new ArrayList<>();
-
-		for (TestDescriptor engineTestDescriptor : request.getRootTestDescriptor().getChildren()) {
-			Optional<String> engineId = engineTestDescriptor.getUniqueId().getEngineId();
-
-			if (!engineId.isPresent()) {
-				continue;
-			}
-
-			TestEngine testEngine = configuration.testEngineRegistry.getTestEngine(engineId.get());
-			AvailableTests availableTestsForEngine = TestDescriptorUtils
-					.getAvailableTests(testEngine, engineTestDescriptor);
-			TestExecutorRequest testExecutorRequest = new TestExecutorRequest(testEngine, engineTestDescriptor,
-					request.getEngineExecutionListener(), request.getConfigurationParameters());
-			List<TestExecution> testExecutionsOfEngine = configuration.testExecutor.execute(testExecutorRequest);
-
-			testExecutions.addAll(testExecutionsOfEngine);
-			availableTests.addAll(availableTestsForEngine.getTestList());
-		}
-
-		dumpTestDetails(availableTests, configuration.reportDirectory);
-		dumpTestExecutions(testExecutions, configuration.reportDirectory);
-	}
-
-	private static void dumpTestExecutions(List<TestExecution> testExecutions, File reportDirectory) {
-		writeReport(new File(reportDirectory, "test-execution.json"), testExecutions);
-	}
-
-	/** Writes the given test details to a report file. */
-	private static void dumpTestDetails(List<TestDetails> testDetails, File reportDirectory) {
-		writeReport(new File(reportDirectory, "test-list.json"), testDetails);
-	}
-
-	private static <T> void writeReport(File file, T report) {
-		try {
-			ReportUtils.writeReportToFile(file, report);
-		} catch (IOException e) {
-			LOGGER.error(e, () -> "Error while writing report to file: " + file);
-		}
+		Preconditions.notNull(internalImpactedTestEngine, "Can't execute request without discovering it first.");
```

Replace with assertion. Otherwise tests would silently fail if this case happens